### PR TITLE
feat: Auto collect words (add #883)

### LINKED ISSUE #883: 自动加入生词本
### Search before asking

- [X] 在 [issues](https://github.com/yetone/openai-translator/issues) 中没有找到类似的内容。
I searched in the [issues](https://github.com/yetone/openai-translator/issues) and found nothing similar.

### feature

chatgpt返还结果后自动加入生词本

### 描述 Motivation

每次点击星号不是很有效率，如果提供自动加入的选项就好了：））））

### Solution

_No response_

### 还有其他内容吗？Anything else?

_No response_

### 你是否愿意提交一份 PR？Are you willing to submit a PR?

- [ ] 我愿意提供 PR! I'm willing to submit a PR!

### PR DESCRIPTION
**Pull request by OverflowCat**
screenshot

## This commit resolves #883 

- [x] A button for auto collection
- [x] Persistence for the setting

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -15,9 +15,9 @@
     "clean": "make clean",
     "test": "vitest test",
     "test:e2e": "playwright test",
-    "lint": "eslint 'src/**/*.{ts,tsx}' --cache",
-    "lint:fix": "eslint --fix 'src/**/*.{ts,tsx}'",
-    "format": "prettier --write 'src/**/*.{js,jsx,ts,tsx,css,md,json}' --config ./.prettierrc --cache"
+    "lint": "eslint \"src/**/*.{ts,tsx}\" --cache",
+    "lint:fix": "eslint --fix \"src/**/*.{ts,tsx}\"",
+    "format": "prettier --write \"src/**/*.{js,jsx,ts,tsx,css,md,json}\" --config ./.prettierrc --cache"
```
Comment (OverflowCat) [inline package.json:20]: Windows has to use double quotes -- see https://stackoverflow.com/a/67273107

File: src/common/translate.ts [hunk 2]
```diff
@@ -499,6 +499,7 @@ export async function translate(query: TranslateQuery) {
             break
     }
 
+    let finished = false // finished can be called twice because event.data is 1. "finish_reason":"stop"; 2. [DONE]
```
Comment (OverflowCat) [inline src/common/translate.ts:502]: Explanation: Before this fix, `setTranslatedText` will be invoked twice

**Review by yetone — APPROVED** [review]
LGTM!

**Comment by yetone** [comment]
image

I tried it out, and it seems like only the first attempt is successful.

**Pull request merged**